Team 7

mwhittaker opened the pull request:
The final report can be found [here](https://github.com/WensiWu/water/blob/master/doc/final/final.pdf). The mid-report can be found [here](https://github.com/WensiWu/water/blob/master/doc/mid/mid.pdf).

mwhittaker committed:
Now you can build any version of simulator!

mwhittaker committed:
Merge remote-tracking branch 'upstream/master'

Commit:
add profile script and initial result

mwhittaker committed:
Cleaning shallow_v2 in Makefile.

mwhittaker committed:
Added .gitignore.

mwhittaker committed:
Added validator.

Now, whenever you run a simulator, it is checked against the reference
simulator. To verify that the validator is working, I made two alternate
implementations of the simulator.

    1. central2d_copy is a 100% copy of central2d
    2. central2d_buggy is a copy of central2d with an intentional bug

When you run `make shallow_copy` and `make shallow_buggy`, only
`shallow_copy` succeeds; `shallow_buggy` aborts early. This is the
intended behaviour!

Commit:
cleared up some dependencies

Commit:
Central2d_v3

mwhittaker committed:
Re-added buggy code. Added v3.

Commit:
omp nested loop bug fixed

Commit:
updated

TJSong committed:
add plotter.py script

USAGE : ./plotter.py [qsub_result ...]

Ex) If you have v1.o1234, v1.o1235, and v2.o1236
    run like : ./plotter.py v1.o* v2.o*
    This will generate timing.pdf file.

TJSong committed:
modify header of plotter.py

Commit:
modify driver.cc to print arguments

Commit:
Update

mwhittaker committed:
Fixed .gitignore bug.

I was ignoring too many files before.

mwhittaker committed:
Removed std::array in favor of flattened vectors.

Commit:
Updated parallel code

Commit:
Updated parallel code

Commit:
Modify timing for N iterations

Commit:
Merge branch 'master' of github.com:WensiWu/water

mwhittaker committed:
Changed Makefile to use icc, not gcc.

mwhittaker committed:
Changed vectors to arrays.

mwhittaker committed:
Changed new/delete to malloc/free.

I think new/delete was even confusing the compiler.

mwhittaker committed:
Finally saw some improvements with the vectorizer!

mwhittaker committed:
Added generic ampl running script.

mwhittaker committed:
Cleaned up timing code.

Commit:
Bugs fixed

mwhittaker committed:
Cleaed up timing

mwhittaker committed:
Added skeleton mid report.

mwhittaker committed:
Added sections.

mwhittaker committed:
Removed crazy tex deps.

mwhittaker committed:
Added section assigments.

mwhittaker committed:
Removed release README.

mwhittaker committed:
Updated README.

mwhittaker committed:
Fixed README typo.

mwhittaker committed:
Ignored VTune output.

TJSong committed:
add figures and timing.tex

mwhittaker committed:
Distinguished ampl qsub runs from non-ampl.

mwhittaker committed:
Removed comments from vec + minor spacing.

mwhittaker committed:
Added some notes on what's vectorized so far.

mwhittaker committed:
More work on vectorization.

mwhittaker committed:
More Makefile magic.

TJSong committed:
modify README

Commit:
reduced use of parallel for

Commit:
Merge branch 'master' of https://github.com/WensiWu/water

ww382 committed:
Added parallelization seciton to the report

ww382 committed:
Merge branch 'master' of https://github.com/WensiWu/water

Commit:
add different types of plots

TJSong committed:
Update README.md

Type 1 and 2 added

mwhittaker committed:
Fixed '#' character in tex.

mwhittaker committed:
Rearranged sections.

mwhittaker committed:
Added software engineering section.

mwhittaker committed:
Added verification section.

mwhittaker committed:
Added software engineering section.

mwhittaker committed:
Revisions.

TJSong committed:
add figures

mwhittaker committed:
Fixed bug in plotting instructions.

*.o* matches more than just outfiles from pbs scripts.

mwhittaker committed:
Minor formatting.

mwhittaker committed:
Added example of non-vectorized code.

mwhittaker committed:
Experimenting with loop ordering for vectorization.

mwhittaker committed:
Added vectorization section.

mwhittaker committed:
Typographical changes.

mwhittaker committed:
Added future work section.

TJSong committed:
add evaluation section to mid report

mwhittaker committed:
Finished intro.

mwhittaker committed:
Minor revisions.

mwhittaker committed:
Added report.

calvinwylie commented:
# 

#  Review from Group 6 (For Group 7)
1. Strengths
- Great work on additional verification. I assume the reference solution
  generated from the serial implementation of the algorithm? Do you get
  any significant differences in the solutions with the floating point
  rounding?
- I liked the discussion on how to nudge the compiler to actually
  vectorize the code. Nice figure.
1. Potential Improvements
- The parallelization strategy used here is a bit vague, specifically on
  how you orchestrated the inter-thread communication.
- This is something our group needs to look into as well, but it would be
  interesting to see exactly why limited_derivs dominates execution time.
1. Additional Comments
- Might want to crunch the hanging sentence at the end of pp.4

sjasonma commented:
From Group 5:

Things you did well:
- The verification part is interesting and might be useful for other groups.
- Good analysis and discussion in Vectorization part.
- The figure illustrates clearly in the evaluation part.

Things you should work on:
- A little more detail in the parallelization part might be useful.
- I believe digging deeper in the domain decomposing will be beneficial.
- You found that the limited_derivs function is the bottleneck, but how do you deal with it?

mwhittaker committed:
Reordered some loops, contemplating vectorization.

Commit:
add my own pbs file

mwhittaker committed:
Copied vectorized code to blocked code.

mwhittaker committed:
Separated max speed and flux computation.

mwhittaker committed:
Work on blocking.

mwhittaker committed:
Fixed max_speed bug.

mwhittaker committed:
Removed const getters.

mwhittaker committed:
Adding static getters.

This will come in handy when we block up data.

mwhittaker committed:
Added indexers.

mwhittaker committed:
Part way through blocking.

mwhittaker committed:
Modified Makefile.

Removed vector flag because I've been running on the head node. I've
also added the restrict flag to make things work.

mwhittaker committed:
Added blocking of flux.

I added blocked flux, but I'm not sure it's correct at all. For some
weird reason, doing anything with fluxes doesn't affect the final
result.

mwhittaker committed:
Added blocked limderiv.

mwhittaker committed:
Work on blocking compute step.

mwhittaker committed:
Blocked corrector.

TJSong committed:
[doc/final] add initial final report structure

TJSong committed:
modify title final.tex

mwhittaker committed:
Redoing blocking.

mwhittaker committed:
Added flat array test.

mwhittaker committed:
Added bare bones block class.

mwhittaker committed:
More work on block.

mwhittaker committed:
More refinment of block interface.

mwhittaker committed:
Constructing blocks.

mwhittaker committed:
Fixed erroneous sizes in block construction.

mwhittaker committed:
Unchecked write back.

mwhittaker committed:
Ported flux.

mwhittaker committed:
Ported more block helpers.

mwhittaker committed:
Added shallow block 2.

mwhittaker committed:
Fixed small bugs.

The functions being generated by macros weren't returning anything, yet
the compiler didn't warn me. I danced with the devil and I got burnt.
What did I expect to happen?

Commit:
#pragma vector aligned added for vectorization

mwhittaker committed:
Finished first stab at blocking.

mwhittaker committed:
Clean-up and removing unneeded functions.

mwhittaker committed:
Fixed erroneous port of flux.

mwhittaker committed:
Swapped loop order in loop.

mwhittaker committed:
Fixed blocking bug!

mwhittaker committed:
Fixed up Makefiles.

mwhittaker committed:
Tidied Makefile.

mwhittaker committed:
Merge branch 'block_dev'

mwhittaker committed:
Clobbered block with block2.

TJSong committed:
add figures

Commit:
added omp for to blocking

mwhittaker committed:
Pointer swapping trick for writeback.

mwhittaker committed:
Stole "optimized" minmod limiter from bindell.

mwhittaker committed:
Finished adding minmon opt.

mwhittaker committed:
Vectorized limdiff.

mwhittaker committed:
Vectorized vec.

mwhittaker committed:
Updated Makefile.

mwhittaker committed:
Tweaked vectorization.

mwhittaker committed:
Added pbs to sweep values of nx.

mwhittaker committed:
Updated parametric sweeping.

mwhittaker committed:
Found optimal block sizes.

mwhittaker committed:
Merge branch 'dev'

mwhittaker committed:
Vectorized block impl.

mwhittaker committed:
Removed redundant copies from block impl.

mwhittaker committed:
Added beast opt!

TJSong committed:
add opt-timing plots

TJSong committed:
update evaluation.tex

TJSong committed:
add nx 1000 timing plots

TJSong committed:
rename timing2.pdf

Commit:
remove frames from driver.cc to measure total time

Commit:
modify time format %e to %f

mwhittaker committed:
Added shallow run-sweep.

TJSong committed:
update evaluation with final timing graphs

mwhittaker committed:
Fixed tex bugs.

TJSong committed:
fix figure ref

ww382 committed:
Updated parallelzation

ww382 committed:
Merge branch 'master' of https://github.com/WensiWu/water

mwhittaker committed:
Fixed my net id :P

mwhittaker committed:
Revisions to some older sections.

mwhittaker committed:
80-character wrapped parallelization section.

mwhittaker committed:
Revised parallelization sec.

mwhittaker committed:
Deleted future work section.

mwhittaker committed:
Added domain decomp section.

mwhittaker committed:
80 character wrapped eval section.

mwhittaker committed:
Revised eval section.

ww382 committed:
added final report

mwhittaker committed:
Updated final report.

ww382 committed:
fixed reference
